Dockerfile

```
@@ -2,16 +2,19 @@ FROM python:3-slim AS builder
 
 RUN mkdir /src
 COPY . /src/
+ENV VIRTUAL_ENV=/opt/venv
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Don't know if this will work - Ignore if it does not ...

```suggestion
RUN source /opt/venv/bin/activate
```
Lets use the inbuilt `activate` script venv includes ... This way if any fixes are made in future python's venv module we get them for free.

---

Thanks for the suggestion, i agree. Being that RUN instructions are independent from each other we are kind of forced to do this:
`RUN . /opt/venv/bin/activate && pip install ...`
It works indeed. I'm adding this, let me know what you think about it!